File: solutions/python/pythagorean-triplet/2/pythagorean_triplet.py
import itertools


def triplets_with_sum(n: int) -> list[list[int]]:
    triplets = []
    for a in range(1, n//3 + 1):
        b = (n**2 - 2 * a * n) // (2 * (n - a))
        c = n - a - b

        if a**2 + b**2 == c**2 and b > 0 and c > 0 and [b, a, c] not in triplets:
            triplets.append([a, b, c])

    return triplets

File: solutions/ruby/pythagorean-triplet/1/pythagorean_triplet.rb
module PythagoreanTriplet
  def self.triplets_with_sum(n)
    triplets = Set.new()

    (1..n/3.floor+1).each {|a|
      b = (n**2 - 2 * a * n) / (2 * (n - a)).floor
      c = n - a - b

      if a**2+b**2==c**2 && b > 0 && c > 0 && b > a
        triplets << [a,b,c]
      end
    }

    triplets.to_a
  end
end